Lib/test/test_threading.py

```diff
@@ -1604,6 +1604,31 @@ def test_interrupt_main_invalid_signal(self):
         self.assertRaises(ValueError, _thread.interrupt_main, signal.NSIG)
         self.assertRaises(ValueError, _thread.interrupt_main, 1000000)
 
+    @threading_helper.reap_threads
+    def test_can_interrupt_tight_loops(self):
+        cont = True
+        started = False
+        iterations = 100_000_000
+
+        def worker():
+            nonlocal iterations
+            nonlocal started
+            started = True
+            while cont:
+                if iterations:
+                    iterations -= 1
+                else:
+                    return
+                pass
+
+        t = threading.Thread(target=worker)
+        t.start()
+        while not started:
+            pass
+        cont = False
+        t.join()
+        self.assertNotEqual(iterations, 0)
```

Comment:
```suggestion
    @threading_helper.reap_threads
    def test_can_interrupt_tight_loops(self):
        cont = True
        started = threading.Event()
        iterations = 100_000_000

        def worker():
            nonlocal iterations
            started.set()
            while cont:
                if iterations:
                    iterations -= 1
                else:
                    return
                pass

        t = threading.Thread(target=worker)
        t.start()
        started.wait()
        cont = False
        t.join()
        self.assertNotEqual(iterations, 0)
```

Reply:
We are testing whether tight loops can be interrupted. Adding all the thread event machinery could undermine that.
I'd rather keep the test as close to the original bug report as possible. A little bit of active waiting shouldn't be a problem.

Reply:
Ok, I don't want to hold this more but note that the event is *outside* the loop, so it should not interfere